### PR TITLE
Improve base path hint rule

### LINKED ISSUE #154: Base URL Hint is misleading for edge cases
If there is only on endpoint available, the hint rule "you could extract this to base url" generates false-positives, for instance: https://raw.githubusercontent.com/zalando-incubator/api-discovery/master/storage/src/main/resources/api/storage-api.yaml

This hint rule should only fire if depths of endpoints is > 1, e.g. if I only have /api and different http methods on it with different parameters, this rule is not violated.

### PR DESCRIPTION
**Pull request by fmueller**
Fixes #154

### CODE REVIEW

**Comment by fmueller** [comment]
Integration Tests are breaking, I'll fix this.

**Comment by fmueller** [comment]
@netme I fixed the integration tests, can you please review it? :)

File: cli/integration_tests/tests.bats [hunk 1]
```diff
@@ -50,9 +50,9 @@
...
-	[ "${lines[10]}" = "[0mMUST violations: 0" ]
-	[ "${lines[11]}" = "SHOULD violations: 0" ]
-	[ "${lines[12]}" = "COULD violations: 0" ]
-	[ "${lines[13]}" = "HINT violations: 1" ]
-	[ "${#lines[@]}" -eq 14 ]
+	[ "${lines[3]}" = "[0mMUST violations: 0" ]
+	[ "${lines[4]}" = "SHOULD violations: 0" ]
+	[ "${lines[5]}" = "COULD violations: 0" ]
+	[ "${lines[6]}" = "HINT violations: 0" ]
+	[ "${#lines[@]}" -eq 7 ]
```
Comment (netme) [inline cli/integration_tests/tests.bats]: Nice!

**Comment by netme** [comment]
👍

**Pull request merged**